### PR TITLE
chore: fix long socket path breaking test on osx

### PR DESCRIPTION
**Pull request by crenshaw-dev**
The handy `t.TempDir()` function makes paths that are so long, they break tests on osx. There's a path character limit for socket files, and apparently `Test_GPG_InitializeGnuPG` relies on a socket file in that temp dir.

### CODE REVIEW

File: util/gpg/gpg_test.go [hunk 1]
```diff
@@ -227,7 +241,7 @@ func Test_ImportPGPKeysFromString(t *testing.T) {
 }
 
 func Test_ValidateGPGKeysFromString(t *testing.T) {
-	initTempDir(t)
+	_ = initTempDir(t)
```
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply util/gpg/gpg_test.go]: Because I got sloppy with the find/replace. 😆  Reverted, thanks for the catch!

**Comment by crenshaw-dev** [comment]
Not sure what's up w/ codecov these days.

**Review by rishabh625 — APPROVED** [review]

**Review by leoluz — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry-picked onto 2.4 so I can run patch tests locally.

**Pull request merged**